### ISSUE TITLE
Prerendering doesn't respect `baseURL`

### ISSUE THREAD

**Issue by gnehs** [nuxt/framework#4581]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.1`
- Package Manager:  `npm@8.5.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

- `npx nuxi init nuxt-app`
- `cd nuxt-app`
- `npm install`
- `NUXT_APP_BASE_URL=/foo npm run generate`

### Describe the bug

When NUXT_APP_BASE_URL is specified, the deployed file does not have index.html
#### not specify NUXT_APP_BASE_URL

```
> npm run generate
> tree ./.output/public 
├── _nuxt
│   ├── entry-79c674ce.mjs
│   ├── entry.93eb12fa.css
│   └── manifest.json
└── index.html

1 directory, 4 files
```
#### specify NUXT_APP_BASE_URL
```
> NUXT_APP_BASE_URL=/foo npm run generate
> tree ./.output/public
.
└── _nuxt
    ├── entry-79c674ce.mjs
    ├── entry.93eb12fa.css
    └── manifest.json

1 directory, 3 files
```
missing index.html :(

### Logs

```shell
> generate
> nuxt generate

Nuxt CLI v3.0.0-rc.1                                                    16:21:28
ℹ Vite client warmed up in 1652ms                                       16:21:34
ℹ Client built in 1674ms                                                16:21:34
ℹ Building server...                                                    16:21:34
✔ Server built in 563ms                                                 16:21:34
✔ Generated public .output/public                                 nitro 16:21:34
ℹ Initializing prerenderer                                        nitro 16:21:34
ℹ Prerendering 1 initial routes with crawler                      nitro 16:21:37
  ├─ / (47ms) (Error: [404] Not Found)                            nitro 16:21:37
✔ You can now deploy .output/public to any static hosting!              16:21:37
```

**Comment by mario-neuhold** [comment]
Also happening in rc.2 and (not sure if it matters) ssr: false. With ssr I get JS heap out of memory exception, not sure if unrelated.

```
-----------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.2`
- Package Manager:  `yarn@1.22.15`
- Builder:          `vite`
- User Config:      `ssr`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------
```

**Comment by daniel-fahl** [comment]
I observed the same behavior. Didn't know that it was connected to app.baseURL config before finding this bugreport.

- Operating System: `Docker on MacOS 12.3.1 M1 Chip`
- Node Version: `v16.14.2`
- Nuxt Version: `3.0.0-rc.1`
- Package Manager: `yarn@1.22.18`
- Builder: `vite`
- User Config: `target: "static", app.baseURL: "/something/"`
- Runtime Modules: -
- Build Modules: `@nuxtjs/strapi, @nuxtjs/tailwindcss`

**Comment by danielroe** [comment]
This should be resolved in nitropack v0.4.3. which you can get with `yarn upgrade nuxt` or `npm upgrade nuxt`.

Let me know if not and I'll reopen.
